poll/poll.py

```diff
@@ -1296,7 +1298,7 @@ def generate_report_data(self, user_state_iterator, limit_responses=None):
         count = 0
         for user_state in user_state_iterator:
             # user_state.state={'submissions_count': 1, 'choices': {u'enjoy': u'Y', u'recommend': u'N', u'learn': u'M'}}
-            choices = user_state.state['choices']
+            choices = user_state.state.get('choices', {})
```

Comment:
I think the main case we need to solve for here is if choices is ``None``. I'm not sure it's possible for the key to be entirely absent. If choices is None, the above code will still resolve to None and the error will occur. So the above should be:

```suggestion
            choices = user_state.state.get('choices') or {}
```

Reply:
Good catch! I've incorporated this change and added a test for it.